src/library/scala/collection/LinearSeqOptimized.scala

```diff
@@ -315,4 +315,7 @@ trait LinearSeqOptimized[+A, +Repr <: LinearSeqOptimized[A, Repr]] extends Linea
     }
     last
   }
+
+  override /*TraversableLike*/
+  def tails: Iterator[Repr] = Iterator.iterate(repr)(_.tail).takeWhile(_.nonEmpty) ++ Iterator(newBuilder.result)
```

Comment:
The existing implementations are implemented with:

https://github.com/scala/scala/blob/403f910996ebd5cbf327e8cc978d03b906b83ea7/src/library/scala/collection/TraversableLike.scala#L804-L807

I wonder if `Iterator(Nil) map (x => (newBuilder ++= x).result)` could simplified to the version above  (which, although simpler, still took me a little while to recognize as `take(0)` 😄 )

Reply:
If i remember correctly, it looks like this in TraversableLike because the only "path" from Traversable[A @uV] to Repr is through newBuilder. Traversable is not constrained to Repr. So `Iterator(Nil) map (x => (newBuilder ++= x).result)` is basically a cast to Iterator[Repr] from Iterator[Traversable[A @uV]]